change pronoun input margin to value for grouped form elements

anthony-hull opened the pull request:
### Details
Increased the margin between the pronoun elements.

### Fixed Issues

$ https://github.com/Expensify/App/issues/5556

### QA Steps

1. View Profile
2. Change preferred pronouns to Self-select
3. The margin between the pronoun form elements is equal to that between First name and Last name


### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
Web
![image](https://user-images.githubusercontent.com/47184118/144756038-7737d45e-55ca-40aa-aaa8-b6e4ef791d58.png)

Tablet mobile web
![image](https://user-images.githubusercontent.com/47184118/144756094-6e2c553d-f9c1-4e0e-ac4f-ec3a86fea928.png)

Mobile Web
![image](https://user-images.githubusercontent.com/47184118/144756130-cb9b2122-d1ad-454a-a317-b092c886d79e.png)

iOS
![image](https://user-images.githubusercontent.com/47184118/144755887-36149956-c6bd-4a42-8bb0-4cd246f5d35c.png)

Android
![image](https://user-images.githubusercontent.com/47184118/144756662-a394cb2b-f176-4e6e-8b2f-6b8b5dfe8318.png)

anthony-hull committed:
change pronoun input margin to value for grouped form elements

parasharrajat reviewed (changes requested):
It increases the gap between select dropdown and next element when there is no custom Pronoun input. Could you please fix that? 
it should not affect the UI when the custom input is hidden. Could you please test it carefully?

anthony-hull commented:
I'm very sorry! You are right, even simple changes need full attention.

anthony-hull committed:
move margin to conditional element

this avoids margin being present when the optional pronouns input filed isn't present

anthony-hull commented:
![image](https://user-images.githubusercontent.com/47184118/144759282-c1ad38d8-ca97-4235-a458-6fed9847d838.png)
![image](https://user-images.githubusercontent.com/47184118/144759307-1a6c8f53-1f0b-437e-9dab-7c19f56e347f.png)

parasharrajat reviewed (approved):
LGTM. 

All yours @puneetlath.

puneetlath reviewed (approved):
Looks good!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1547012777) to staging by @puneetlath in version: 1.1.17-8 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1555704648) to production by @roryabraham in version: 1.1.18-3 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
